### PR TITLE
Create modal changelog item

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

For https://github.com/uswds/uswds/pull/5315

## Preview link

Preview link:
[Modal changelog →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-modal-refactor-changelog/components/modal/#latest-updates)

### CODE REVIEW

File: _data/changelogs/component-modal.yml
Comment (mejiaj) [inline _data/changelogs/component-modal.yml:5]: Noting that we'll need to add date before merge.

File: _data/changelogs/component-modal.yml
Comment (mejiaj) [inline _data/changelogs/component-modal.yml:12]: Need to update version before merge.

**Review by amyleadem — CHANGES_REQUESTED** [review]
Generally looks good. I added a couple of small suggestions in the comments:

File: _data/changelogs/component-modal.yml
Comment (mahoneycm) [reply _data/changelogs/component-modal.yml]: To me, the main "change/improvement" revolves around the failsafe measures, and the JS multi-initialization prevention is a result of that. What do you think of this change? 

```yml
  summary: Updated JavaScript to include fallbacks and error messages when initialization cannot be completed.
  summaryAdditional: Additionally, this improved code consistency and prevented errors when trying to initialize modal twice.
```

Or would it be better practice to address the problem statement first like your example?

Comment (amyleadem) [reply _data/changelogs/component-modal.yml]: I like the wording on that update, @mahoneycm. Happy to defer to you about identifying the primary benefit here.

**Review by amyleadem — APPROVED** [review]

**Pull request merged**
